server/src/weaverbird/pipeline/pipeline.py

```diff
@@ -280,14 +279,22 @@ def _remove_empty_elements(data: Any) -> Any:
                 if (cleaned := _remove_empty_elements(v)) is not None:
                     data_transformed[k] = cleaned
 
-        return data_transformed or None
+                if cleaned == [] and k in ["$or", "$nor", "$and"]:
+                    data_transformed[k] = None
```

Comment:
Shouldn't it just be omitted in case the key is `$or`, `$and` or `$nor` ? I don't think `$and: null` is valid mongo syntax

Reply:
🤔 We will never have the case of `$and: null` at the end of the transformation !
See the updated test, please!
```python
    assert remove_void_conditions_from_mongo_steps(
        [
            {
                "$match": {
                    "$ne": None,
                    "$and": [
                        {"$or": [{"property2": "value2"}, {"property3": {}}]},
                        {"$nor": [{"property4": {}}, {"property5": {}}]},
                    ],
                    "$or": [],
                    "$nor": [
                        {
                            "property9": {
                                "$and": [
                                    {
                                        "property7": {},
                                    }
                                ]
                            }
                        }
                    ],
                }
            }
        ]
    ) == [{"$match": {"$and": [{"$or": [{"property2": "value2"}]}], "$ne": None}}]
```

Reply:
Woops sorry :facepalm: